### PR TITLE
FailureDetector phi metrics

### PR DESCRIPTION
**Pull request by rhuffy**
Add new metrics to track details of the FailureDetector.

### CODE REVIEW

File: src/java/com/palantir/cassandra/metrics/FailureDetectorMetrics.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
+
+public class FailureDetectorMetrics
+{
+    public static void register(InetAddress ep, ArrivalWindow window)
+    {
+        Metrics.register(createMetricName(ep, "phi"), (Gauge<Double>) window::getLastReportedPhi);
+        Metrics.register(createMetricName(ep, "interval_histogram"), new Histogram(window));
+        Metrics.register(createMetricName(ep, "last_interval"), (Gauge<Long>) window::getLastInterval);
```
Comment (inespot) [inline src/java/com/palantir/cassandra/metrics/FailureDetectorMetrics.java]: Nit changes: considering how they appear in grafana we probably wanna rename this to sth like PhiLastInterval 
We should also match the CamelCase format of other metrics for consistency

File: src/java/com/palantir/cassandra/metrics/FailureDetectorMetrics.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    private static CassandraMetricsRegistry.MetricName createMetricName(InetAddress ep, String name)
+    {
+        String endpoint = InetAddresses.toAddrString(ep);
+        String groupName = FailureDetectorMetrics.class.getPackage().getName();
+
+        StringBuilder mbeanName = new StringBuilder();
+        mbeanName.append(groupName).append(":");
+        mbeanName.append("type=FailureDetector");
+        mbeanName.append(",endpoint=").append(endpoint);
+        mbeanName.append(",name=").append(name);
```
Comment (wi11dey) [inline src/java/com/palantir/cassandra/metrics/FailureDetectorMetrics.java:58]: nit: i think might be more readable as String.format

File: src/java/com/palantir/cassandra/metrics/ReadOnlyHistogram.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Snapshot;
+
+public class ReadOnlyHistogram extends Histogram
+{
+
+    private final Supplier<Snapshot> snapshotSupplier;
+
+    private int size = -1;
```
Comment (wi11dey) [inline src/java/com/palantir/cassandra/metrics/ReadOnlyHistogram.java]: nit: default to 0 for sidecar

**Review by wi11dey — DISMISSED** [review]

**Review by wi11dey — DISMISSED** [review]

**Review by wi11dey — APPROVED** [review]

**Comment by rhuffy** [comment]
👍

**Pull request merged**